arrow-array/src/array/primitive_array.rs

```diff
@@ -299,6 +299,17 @@ impl<T: ArrowPrimitiveType> PrimitiveArray<T> {
     /// # Panics
     ///
     /// Panics if [`Self::try_new`] returns an error
+    ///
+    /// ```
+    /// # use arrow_array::Int32Array;
+    /// # use arrow_array::types::Int32Type;
+    /// # use arrow_buffer::NullBuffer;
+    /// // [1, 2, 3, 4]
+    /// let array = Int32Array::new(vec![1, 2, 3, 4].into(), None);
+    /// // [1, null, 3, 4]
+    /// let nulls = NullBuffer::from(vec![true, false, true, true]);
```

Comment:
It is unfortunate that arrow went with the name null buffer, and then made true mean not null. But hey ho

Reply:
BTW could this be 

```suggestion
    /// let nulls = NullBuffer::from(&[true, false, true, true]);
```

And potentially avoid the allocation? I see you added that `From` impl 🤔

Reply:
For `From<&[bool]>`, looks like it still does allocation.

Reply:
It would eliminate the allocation for the Vec, although the compiler might be smart enough to do that anyway, will update